assignments/prototypes.js

```diff
@@ -10,38 +10,80 @@
   
 /*
   === GameObject ===
-  * createdAt
-  * dimensions (These represent the character's size in the video game)
-  * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
+   createdAt
+   dimensions (These represent the character's size in the video game)
+   destroy() // prototype method -> returns the string: 'Object was removed from the game.'
 */
 
+function GameObject(gameAttributes) {
+  this.createdAt = gameAttributes.createdAt;
+  this.dimensions = gameAttributes.dimensions;
+}
+
+GameObject.prototype.destroy = function() {
+  return `${this.name} was removed from the game.`;
+} ;
+
+
+ 
 /*
   === CharacterStats ===
-  * healthPoints
-  * name
-  * takeDamage() // prototype method -> returns the string '<object name> took damage.'
-  * should inherit destroy() from GameObject's prototype
+   healthPoints
+   name
+   takeDamage() // prototype method -> returns the string '<object name> took damage.'
+   should inherit destroy() from GameObject's prototype
 */
 
+function CharacterStats(charAttributes) {
+  GameObject.call(this, charAttributes);
+  this.healthPoints = charAttributes.healthPoints;
+  this.name = charAttributes.name;
+}
+
+CharacterStats.prototype = Object.create(GameObject.prototype);
+
+CharacterStats.prototype.takeDamage = function() {
+  return `${this.name} took damage.`;
+};
+
+
+
+
 /*
   === Humanoid (Having an appearance or character resembling that of a human.) ===
-  * team
-  * weapons
-  * language
-  * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
-  * should inherit destroy() from GameObject through CharacterStats
-  * should inherit takeDamage() from CharacterStats
+   team
+   weapons
+   language
+   greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+   should inherit destroy() from GameObject through CharacterStats
+   should inherit takeDamage() from CharacterStats
 */
+
+function Humanoid(hnAttributes) {
```

Comment:
this parameter totally works, but just so you are aware, you can name the parameters the same thing for different functions/classes if they make sense